src/SDKs/Azure.Base/data-plane/Azure.Base/Diagnostics/HttpPipelineEventSource.cs

```diff
@@ -1,79 +1,193 @@
+        private void RequestContent(string correlationId, byte[] content)
+        {
+            WriteEvent(RequestContentEvent, correlationId, content);
+        }
+
+        [Event(ResponseEvent, Level = EventLevel.Informational)]
+        private void Response(string correlationId, int status, string headers)
+        {
+            WriteEvent(ResponseEvent, correlationId, status, headers);
+        }
+
+
+        [Event(ResponseContentEvent, Level = EventLevel.Verbose)]
+        private void ResponseContent(string correlationId, byte[] content)
+        {
+            WriteEvent(ResponseContentEvent, correlationId, content);
+        }
+
+        [Event(ErrorResponseEvent, Level = EventLevel.Error)]
+        public void ErrorResponse(string correlationId, int status, string headers)
+        {
+            WriteEvent(ErrorResponseEvent, correlationId, status, headers);
+        }
+
+        [Event(ErrorResponseContentEvent, Level = EventLevel.Informational)]
+        private void ErrorResponseContent(string correlationId, byte[] content)
+        {
+            WriteEvent(ErrorResponseContentEvent, correlationId, content);
+        }
+
+        [Event(RequestDelayEvent, Level = EventLevel.Warning)]
+        private void ResponseDelayCore(long delayMilliseconds)
+        {
+            if (IsEnabled(EventLevel.Warning, EventKeywords.None))
+            {
+                WriteEvent(RequestDelayEvent, delayMilliseconds);
+            }
+        }
+
+        private static async Task<byte[]> FormatContentAsync(Stream responseContent)
+        {
+            using (var memoryStream = new MemoryStream())
+            {
+                await responseContent.CopyToAsync(memoryStream);
+
+                // Rewind the stream
+                responseContent.Position = 0;
+
+                return FormatContent(memoryStream.ToArray());
             }
         }
 
-        [Event(LOG_RESPONSE, Level = EventLevel.Informational)]
-        void ProcessingResponse(string response)
+        private static async Task<byte[]> FormatContentAsync(HttpPipelineRequestContent requestContent, CancellationToken cancellationToken)
         {
-            if (IsEnabled(EventLevel.Informational, EventKeywords.None)) {
-                WriteEvent(LOG_RESPONSE, response);
+            using (var memoryStream = new MemoryStream())
+            {
+                await requestContent.WriteTo(memoryStream, cancellation: cancellationToken).ConfigureAwait(false);
+
+                return FormatContent(memoryStream.ToArray());
             }
         }
 
-        [Event(LOG_DELAY, Level = EventLevel.Warning)]
-        void ResponseDelayCore(long delayMilliseconds)
+        private static byte[] FormatContent(byte[] buffer)
         {
-            if (IsEnabled(EventLevel.Warning, EventKeywords.None)) {
-                WriteEvent(LOG_DELAY, delayMilliseconds);
+            var count = Math.Min(buffer.Length, MaxEventPayloadSize);
```

Comment:
nit: what type is count? Is it signed or not?

Reply:
That's why we need the style guideline ;)